api/annotate/annotate.go

```diff
@@ -0,0 +1,294 @@
+	ServiceNameAnnotation     = "logz.io/service-name"
+)
+
+// ResourceAnnotateRequest is the JSON body of the POST request
+// It contains the name, controller_kind, namespace, and log type of the resource
+// name: name of the resource
+// controller_kind: kind of the resource (deployment or statefulset)
+// namespace: namespace of the resource
+// log_type: desired log type
+// container_name: name of the container associated with the request
+// service_name: the desired service name for the application, should delete instrumentation if this filed is empty
+type ResourceAnnotateRequest struct {
+	Name           string `json:"name"`
+	Namespace      string `json:"namespace"`
+	ControllerKind string `json:"controller_kind"`
+	LogType        string `json:"log_type,omitempty"`
+	ContainerName  string `json:"container_name"`
+	ServiceName    string `json:"service_name,omitempty"`
+}
+
+// ResourceAnnotateResponse is the data structure for the custom resource
+// the response will contain a list of these fields
+// name: the name of the custom resource
+// namespace: the namespace of the custom resource
+// controller_kind: the kind of the controller that created the custom resource
+// log_type: the log type of the application that the container belongs to
+// service_name: the updated service name
+
+type ResourceAnnotateResponse struct {
+	Name           string  `json:"name"`
+	Namespace      string  `json:"namespace"`
+	ControllerKind string  `json:"controller_kind"`
+	ServiceName    *string `json:"service_name"`
+	ContainerName  string  `json:"container_name"`
+	LogType        *string `json:"log_type"`
+}
+
+func UpdateResourceAnnotations(w http.ResponseWriter, r *http.Request) {
+	logger := api.InitLogger()
+	// Decode JSON body
+	var resource ResourceAnnotateRequest
+	err := json.NewDecoder(r.Body).Decode(&resource)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Get the Kubernetes config
+	config, err := api.GetConfig()
+	if err != nil {
+		logger.Error(api.ErrorKubeConfig, err)
+		http.Error(w, api.ErrorKubeConfig, http.StatusInternalServerError)
+		return
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	// Create a dynamic client
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		logger.Error(api.ErrorDynamic, zap.Error(err))
+		http.Error(w, api.ErrorDynamic+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	// instrumented application crd scheme
+	gvr := schema.GroupVersionResource{
+		Group:    api.ResourceGroup,
+		Version:  api.ResourceVersion,
+		Resource: api.ResourceInstrumentedApplication,
+	}
+	// Validate input before updating resources to avoid changing resources and retuning an error
+	validRequest := isValidResourceAnnotateRequest(resource)
+	// if one of the requests is invalid, return an error
+	if !validRequest {
```

Comment:
Why not just combine it to something like -

```go
if !isValidResourceAnnotateRequest(resource) {
```

I don't see that we use this variable in another place